File: coding/11th March 2026/Complement of Base 10 Integer.java
//Complement of Base 10 Integer
class Solution 
{
    public int bitwiseComplement(int n) 
    {
        String bin = Integer.toBinaryString(n);
        int res = 0;
        for(char ch : bin.toCharArray())
        {
            if(ch == '0')
                res = (res * 2) + 1;
            else
                res = res * 2;
        }
        return res;
    }
}

File: Coding in Python/12th March 2026/Complement of Base 10 Integer.py
#Complement of Base 10 Integer
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        s = bin(n)[2:]
        res = 0
        for ch in s:
            if ch == '0':
                res = (res * 2) + 1
            else:
                res = res * 2
        return res
